### PR TITLE
Tests disable cleaner

### PR DESCRIPTION
**Pull request by pdebski21**

### CODE REVIEW

File: tests/functional/pyocf/types/cache.py [hunk 1]
```diff
@@ -248,7 +248,10 @@ def __init__(
         self.cores = []
 
     def start_cache(
-        self, init_mngmt_queue=True, init_default_io_queue=True, locked: bool = False,
+        self,
+        init_mngmt_queue=True,
+        init_default_io_queue=True,
+        locked: bool = False,
```
Comment (robertbaldyga) [inline tests/functional/pyocf/types/cache.py:254]: Major part of this commit is actually a refactoring, not really related to the commit title. Please extract the refactoring part to the separate commit.

**Review by robertbaldyga — CHANGES_REQUESTED** [review]

**Review by jfckm — DISMISSED** [review]

File: tests/functional/tests/management/test_disable_cleaner.py [hunk 2]
```diff
@@ -57,7 +80,61 @@ def test_load_cleaner_disabled(pyocf_ctx):
...
+    io = vol.new_io(
+        cache.get_default_queue(), S.from_sector(3).B, write_data.size, IoDir.WRITE, 0, 0
+    )
+    io.set_data(write_data)
+
+    cmpl = OcfCompletion([("err", c_int)])
+    io.callback = cmpl.callback
+    io.submit()
+    cmpl.wait()
+    assert 0 == int(cmpl.results["err"])
```
Comment (robertbaldyga) [inline tests/functional/tests/management/test_disable_cleaner.py:103]: This test objective is to just check if disable_cleaner option is preserved between loads. No data consistency check is needed for that.

**Review by robertbaldyga — CHANGES_REQUESTED** [review]

**Review by jfckm — APPROVED** [review]

File: tests/functional/tests/management/test_disable_cleaner.py (hunk 2 above)
Comment (pdebski21) [reply tests/functional/tests/management/test_disable_cleaner.py]: I removed checking data consistency part with io

**Review by jfckm — APPROVED** [review]

File: tests/functional/tests/management/test_disable_cleaner.py [hunk 3]
```diff
@@ -57,7 +79,33 @@ def test_load_cleaner_disabled(pyocf_ctx):
...
     """
-    pass
+    cache_device = RamVolume(S.from_MiB(50))
+    core_device = RamVolume(S.from_MiB(10))
+
+    cache = Cache.start_on_device(cache_device, disable_cleaner=True)
+    core = Core.using_device(core_device)
+
+    cache.add_core(core)
+    vol = CoreVolume(core, open=True)
```
Comment (jfckm) [inline tests/functional/tests/management/test_disable_cleaner.py:89]: you don't need `CoreVolumes` anymore since you're not submitting any IO

File: tests/functional/tests/management/test_disable_cleaner.py [hunk 4]
```diff
@@ -57,7 +79,33 @@ def test_load_cleaner_disabled(pyocf_ctx):
...
+
+    cache.stop()
+
+    cache = Cache.load_from_device(cache_device, open_cores=False, disable_cleaner=True)
+
+    cache.add_core(core, try_add=True)
+    vol = CoreVolume(core, open=True)
```
Comment (jfckm) [inline tests/functional/tests/management/test_disable_cleaner.py:96]: ditto

**Review by jfckm — CHANGES_REQUESTED** [review]

File: tests/functional/tests/management/test_disable_cleaner.py [hunk 5]
```diff
@@ -4,9 +4,17 @@
 #
 
 import pytest
+from pyocf.types.volume import RamVolume
+from pyocf.types.cache import Cache, CacheMetadataSegment
+from pyocf.types.core import Core
+from pyocf.types.volume_core import CoreVolume
+from pyocf.types.data import Data
+from pyocf.types.io import IoDir
```
Comment (jfckm) [inline tests/functional/tests/management/test_disable_cleaner.py:12]: `Data` and `IoDir` is not needed anymore, `CoreVolume` won't be if you remove them

**Review by jfckm — CHANGES_REQUESTED** [review]

**Review by robertbaldyga — APPROVED** [review]
